### PR TITLE
docs: Add ThemeProvider and colors

### PR DESCRIPTION
**Pull request by zero0Halo**
chore(Page): Using ThemeProvider from Anchor

chore(docs): renamed theme.tsx to colors.tsx

chore(docs): made a new subfolder, theme

placed colors.tsx and theme-provider.mdx in it

docs(SideNavigation): Adjusted paths for above changes

docs(ColorSwatchFamily): changed h2 to h3

docs(ColorSwatchFamilyList): extracted text and placed it in colors.mdx

docs: Updated all links to /theme to /theme/colors

docs: Converted colors.tsx to colors.mdx to keep formatting consistent

docs(theme/theme-provider): Heavy editing of content, more detailed explanations of RootTheme

chore(docs): added typing entries for Col, ModalProvider, Row and RootTheme

docs(Page): imported anchor components from anchor library rather than root source

chore(Page): code cleanup

feat(Page): added anchors to all header tags to allow for links to different sections

fix(ModalExample): import styled from xstyled/styled-components rather than styled-components

fix(sections): removed trailing / from theme paths

docs: updated links to point to either colors or theme-provider or theme-provider#colors when necessary

**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config
- [ ] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

Adds documentation on both the `ThemeProvider` (and `RootTheme`) and `colors`. Also includes many small tweaks for linking between files with these new pieces of documentation.

### CODE REVIEW

File: docs/src/components/Navigation/SideNavigation/sections.ts [hunk 1]
```diff
@@ -94,8 +94,12 @@ export const sections = [
         pattern: '/theme',
         links: [
             {
-                title: 'RMN',
-                path: '/theme/',
+                title: 'colors',
+                path: '/theme/colors',
+            },
+            {
+                title: 'ThemeProvider',
```
Comment (brandonlilly) [inline docs/src/components/Navigation/SideNavigation/sections.ts:101]: I think there's value in an `Overview` page for the Theme. I don't know if that's what this PR is intending to accomplish, and I'm merely suggesting a title change, or if that's still forthcoming.

File: docs/src/pages/theme/theme-provider.mdx [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import { Typography } from '@retailmenot/anchor';
+import { Link } from 'gatsby';
+
+# Theme Provider
+
+The `ThemeProvider` is what makes **Anchor** components look the way they do. It is
+from the <a target="_blank" href="https://www.smooth-code.com/open-source/xstyled/">xstyled</a> system,
```
Comment (brandonlilly) [inline docs/src/pages/theme/theme-provider.mdx]: >It is from the xstyled system,

I think the wording here is a bit clunky. I'd probably say that
 - Anchor depends on having a `theme` to read it's values from
 - And the `ThemeProvider` component is what provides the theme context to all styled components
 - The `ThemeProvider` export from `anchor`, `@xstyled/styled-components` and `styled-components` are all the same, so any of them will work interchangeably

File: docs/src/pages/theme/theme-provider.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
+import { Typography } from '@retailmenot/anchor';
+import { Link } from 'gatsby';
+
+# Theme Provider
+
+The `ThemeProvider` is what makes **Anchor** components look the way they do. It is
```
Comment (brandonlilly) [inline docs/src/pages/theme/theme-provider.mdx]: I think we should go into the details of how these theme values can actually be used in your component css and props. For a theme overview page perhaps.

File: docs/src/components/Navigation/SideNavigation/sections.ts (hunk 1 above)
Comment (zero0Halo) [reply docs/src/components/Navigation/SideNavigation/sections.ts:101]: I'd say forthcoming. Lets talk about what would be involved.

File: docs/src/pages/theme/theme-provider.mdx (hunk 2 above)
Comment (zero0Halo) [reply docs/src/pages/theme/theme-provider.mdx]: Heh, clunky. You'd make a good editor. Updated.

File: docs/src/pages/theme/theme-provider.mdx (hunk 3 above)
Comment (zero0Halo) [reply docs/src/pages/theme/theme-provider.mdx]: I agree. I'd like to collab with you on this.

**Pull request closed without merging**